### DISCUSSION TITLE
how to use register_multisession function to align recordings with large translations

### DISCUSSION THREAD

**Discussion by lsainzvillalba** [flatironinstitute/CaImAn#1173] [Q&A]
I am using the alignment function across sessions to identify unique components (neurons) across days of Ca 2p imaging recordings of 512 px x 512 px. (register_multisession function) but as you see in the examples the alignment does not work properly... (In blue background ground truth for a cell highlighted in green, in yellowish background what the algorithm does for the same cell highlighted in orange) 
![Figure_1](https://github.com/flatironinstitute/CaImAn/assets/78734663/02971ff1-547d-4e91-beb5-c35c68e21308)
![Figure_2](https://github.com/flatironinstitute/CaImAn/assets/78734663/0757e2b1-68a8-4e10-9fbd-e6c7d739e074)
![Figure_3](https://github.com/flatironinstitute/CaImAn/assets/78734663/e62b8bec-bd0e-40d1-8499-c2bb88059a97)
![Figure_4](https://github.com/flatironinstitute/CaImAn/assets/78734663/69f1c02e-5243-41e8-ab8a-6be1aab98e6b)

Any thoughts on how to tweak register_multisession parameters like thres_cost or max_dist and others to allow alignment when large translations have happened (~300 px)?

**Comment by EricThomson** [comment]
Thanks for this question: I will look into this more when I get some time. I do know that with really large shifts, it can be tricky to get it right -- we have gotten questions about this before.

**Comment by EricThomson** [comment]
@lsainzvillalba sorry for the delay: will you or someone from your lab be at the workshop -- it would be useful to look at this in person maybe 😃

**Reply by lsainzvillalba** [reply]
When and where is the workshop? I am afraid we are not registered. If possible we could arrange a zoom meeting to discuss it. And thank you in advance for your time.

**Reply by EricThomson** [reply]
Dang the workshop is at SFN coming up. Maybe next time :smile: 

Can you possibly share needed data for registering two sessions (templates and spatial footprints for two tricky sessions)? I may not get to this until after SFN, but would like to work at it.

**Reply by lsainzvillalba** [reply]
Sure, let me gather some sessions and I will let you know...

**Reply by lsainzvillalba** [reply]
@EricThomson , so I was busy with some deadlines but here is an example of two sessions from the same subject in different days. spatial and templates. Thank you for taking time on this... I just saw that I cant upload the files - I have sent you an email via wetransfer
Best,

**Reply by EricThomson** [reply]
@lsainzvillalba I am finally crawling out of 2023 and looking at things, so will get to this soon. Your issue is a high priority, I will get to it near the end of next week, and contact you here when I have questions. 

**edit**: via email it seems Laura has found a solution, but it may be something we want to integrate into Caiman eventually. I'll bring this over to issues/PRs once we've got it sorted.
